ms2deepscore/models/load_model.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Tuple, Union
+import h5py
+import json
+from tensorflow.python.keras.saving import hdf5_format
+
+from ms2deepscore.SpectrumBinner import SpectrumBinner
+from .SiameseModel import SiameseModel
+
+
+def load_model(filename: Union[str, Path]) -> SiameseModel:
+    """
+    Load a MS2DeepScore model (SiameseModel) from file.
+
+    For example:
+
+    .. code-block:: python
+
+        from ms2deepscore.models import load_model
+        model = load_model("model_file_xyz.hdf5")
+
+    Parameters
+    ----------
+    filename
+        Filename. Expecting saved SiameseModel.
+
+    """
+    with h5py.File(filename, mode='r') as f:
+        binner_json = f.attrs['spectrum_binner']
+        keras_model = hdf5_format.load_model_from_hdf5(f)
+
+    # Reconstitute spectrum_binner
+    binner_dict = json.loads(binner_json)
+    spectrum_binner = SpectrumBinner(binner_dict["number_of_bins"],
```

Comment:
Would `SpectrumBinner(**binner_dict)` not work?

Reply:
Unfortunately that doesn't work. the `binner_dict` contains the entire `SpectrumBinner`, but only few things are expected in the first init step. (we can probably make this nicer at some point but I will leave it as is for now)